src/composables/gtag.ts

```diff
@@ -0,0 +1,36 @@
+export const useMaevsiGtag = () => {
+  const {
+    gtag,
+    initialize: initializeGtag,
+    disableAnalytics,
+    enableAnalytics,
+  } = useGtag()
+  const cookieControl = useCookieControl()
+
+  if (cookieControl.cookiesEnabledIds.value?.includes(GTAG_COOKIE_ID)) {
+    initializeGtag()
+  }
+
+  watch(cookieControl.cookiesEnabledIds, (current, previous) => {
+    if (
+      !previous?.includes(GTAG_COOKIE_ID) &&
+      current?.includes(GTAG_COOKIE_ID)
+    ) {
+      initializeGtag()
+      enableAnalytics()
+      gtag('consent', 'update', {
```

Comment:
Further research is needed to confirm if the implementation of Google's new consent mode is correct. I've added a ticket to our task board.